submissions/crealime/oop-exercise/index.js

```diff
@@ -0,0 +1,143 @@
+}
+
+const dog = new Dog('Marley', 'male', 4)
+const cat = new Cat('Sophie', 'female', 4)
+const man = new Human('Harry', 'Taylor', 'male', 2, 2, 'Hello everyone!')
+const woman = new Human('Emily', 'Brown', 'female', 2, 2, 'Hi all!')
+const pirate = new Human('John', 'Silver', 'male', 1, 2, 'Give no quarter!')
+const catWoman = new CatWoman('Selina', 'Kyle')
+
+dog.friends.push(cat, man)
+cat.friends.push(dog, woman, catWoman)
+man.friends.push(dog, woman)
+woman.friends.push(cat, man)
+catWoman.friends.push(cat)
+
+const secretSociety = []
+const allInhabitant = [dog, cat, man, woman, catWoman, pirate]
+
+function addToSociety(society, member) {
+  society.push(member.id)
+}
+
+addToSociety(secretSociety, catWoman)
+addToSociety(secretSociety, pirate)
+
+function isInSociety(society, members) {
+  return members.filter(member => society.includes(member.id)).map(member => member.name)
+}
+
+// ======== OUTPUT ========
+/* Use print(message) for output.
+   Default tag for message is <pre>. Use print(message,'div') to change containing element tag.
+
+   Message can contain HTML markup. You may also tweak index.html and/or styles.css.
+   However, please, REFRAIN from improving visuals at least until your code is reviewed
+   so code reviewers might focus on a single file that is index.js.
+   */
+
+/* Print examples:
+   print('ABC');
+   print('<strong>ABC</strong>');
+   print('<strong>ABC</strong>', 'div');
+
+   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
+
+function getFullInfo(inhabitant) {
+  const getAcquaintance = `This is ${inhabitant.getFullName()} and ${inhabitant.gender === 'male' ? 'he' : 'she'} is a ${inhabitant.species}.`
+  const getGender = `${inhabitant.gender === 'male' ? 'His' : 'Her'} gender is ${inhabitant.gender}.`
+  const getLimbs = `${inhabitant.gender === 'male' ? 'He' : 'She'} have ${inhabitant.hands || 0} ${inhabitant.hands === 1 ? 'hand' : 'hands'} and ${inhabitant.legs || 0} ${inhabitant.legs === 1 ? 'leg' : 'legs'}.`
+  const getFriends = `${inhabitant.name} have ${inhabitant.friends.length} ${inhabitant.friends.length === 1 ? 'friend' : 'friends'}${inhabitant.friends.length > 0 ? ': ' + inhabitant.friends.map(friend => friend.name).join(', ') : ''}.`
+  const getSaying = `${inhabitant.gender === 'male' ? 'He' : 'She'} usually says hello like this: ${inhabitant.saying}`
+
+  return `${getAcquaintance} ${getGender} ${getLimbs} ${getFriends} ${getSaying}`
+}
+
+allInhabitant.forEach(inhabitant => {
+  print(getFullInfo(inhabitant))
+})
+
+function getMembersOfSociety() {
+  const names = isInSociety(secretSociety, allInhabitant)
+  let textAfterNames = ''
+  if (names.length === 1) textAfterNames = 'in a secret society...'
+  if (names.length > 1) textAfterNames = 'are in a secret society...'
```

Comment:
What are other possible outcomes?
Would a ternary operator do the job?

(both questions aren't rhetoric)

Reply:
Thanks! I did not understand right away. The array may be empty. Ternary operator, really more convenient. Done.